sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/QueryPlanningTracker.scala

```diff
@@ -161,3 +166,8 @@ class QueryPlanningTracker {
   }
 
 }
+
+object NoopTracker extends QueryPlanningTracker {
```

Comment:
why is this needed?

Reply:
ic - you just don't want to deal with the Option ?

Reply:
Yea, you're right.